plugins/common/opcua/client.go

```diff
@@ -63,30 +61,53 @@ func (o *OpcUAClientConfig) Validate() error {
 }
 
 func (o *OpcUAClientConfig) validateOptionalFields() error {
-	validFields := []string{"DataType"}
-	return choice.CheckSlice(o.OptionalFields, validFields)
+	for i, field := range o.OptionalFields {
+		switch field {
+		case "DataType":
+			// Valid optional field
+		default:
+			return fmt.Errorf("invalid optional field %q at index %d, expected one of: [DataType]", field, i)
+		}
+	}
+	return nil
 }
 
 func (o *OpcUAClientConfig) validateEndpoint() error {
 	if o.Endpoint == "" {
-		return errors.New("endpoint url is empty")
+		return &EndpointError{
+			Endpoint: o.Endpoint,
+			Err:      fmt.Errorf("%w: endpoint URL is empty", ErrInvalidEndpoint),
+		}
 	}
 
 	_, err := url.Parse(o.Endpoint)
 	if err != nil {
-		return errors.New("endpoint url is invalid")
+		return &EndpointError{
+			Endpoint: o.Endpoint,
+			Err:      fmt.Errorf("%w: %w", ErrInvalidEndpoint, err),
+		}
 	}
 
 	switch o.SecurityPolicy {
-	case "None", "Basic128Rsa15", "Basic256", "Basic256Sha256", "auto":
+	case "None", "Basic128Rsa15", "Basic256", "Basic256Sha256", "Aes128_Sha256_RsaOaep", "Aes256_Sha256_RsaPss", "auto":
+		// Valid security policy
 	default:
-		return fmt.Errorf("invalid security type %q in %q", o.SecurityPolicy, o.Endpoint)
+		validPolicies := []string{"None", "Basic128Rsa15", "Basic256", "Basic256Sha256", "Aes128_Sha256_RsaOaep", "Aes256_Sha256_RsaPss", "auto"}
+		return &SecurityError{
+			Policy: o.SecurityPolicy,
+			Err:    fmt.Errorf("%w: unknown security policy %q, expected one of: %v", ErrInvalidSecurityPolicy, o.SecurityPolicy, validPolicies),
+		}
 	}
 
 	switch o.SecurityMode {
 	case "None", "Sign", "SignAndEncrypt", "auto":
+		// Valid security mode
 	default:
-		return fmt.Errorf("invalid security type %q in %q", o.SecurityMode, o.Endpoint)
+		validModes := []string{"None", "Sign", "SignAndEncrypt", "auto"}
+		return &SecurityError{
+			Mode: o.SecurityMode,
+			Err:  fmt.Errorf("%w: unknown security mode %q, expected one of: %v", ErrInvalidSecurityMode, o.SecurityMode, validModes),
+		}
```

Comment:
I don't think we should add the valid modes in the error message. While in general this is nice, this would mean _loooong_ texts in certain places and the valid options should be documented in either the sample-config or Readme.